Code/GraphMol/DetermineBonds/DetermineBonds.cpp

```diff
@@ -18,9 +18,104 @@
 #include <unordered_map>
 #include <boost/graph/adjacency_list.hpp>
 #include <boost/graph/max_cardinality_matching.hpp>
+#include <boost/multiprecision/cpp_int.hpp>
 
 typedef boost::adjacency_list<boost::vecS, boost::vecS, boost::undirectedS>
     Graph;
+using boost::multiprecision::uint1024_t;
+
+namespace {
+
+
+// see http://phrogz.net/lazy-cartesian-product
+template <typename T>
+struct LazyCartesianProduct {
+  std::vector<std::vector<T>> listOfLists;
```

Comment:
Please prefix the names of all data members with `d_` to make the rest of the code a bit easier to read